packages/react/src/Pagination/Pagination.tsx

```diff
@@ -97,89 +54,59 @@ export const Pagination = forwardRef(
     }: PaginationProps,
     ref: ForwardedRef<HTMLElement>,
   ) => {
-    const [currentPage, setCurrentPage] = useState(page)
+    const Link = linkComponent
 
     // Get array of page numbers and / or spacers
-    const range = useMemo(
-      () => getRange(currentPage, totalPages, maxVisiblePages),
-      [currentPage, totalPages, maxVisiblePages],
-    )
-
-    const onChangePage = (newPage: number) => {
-      if (onPageChange !== undefined) {
-        onPageChange(newPage)
-      }
-      setCurrentPage(newPage)
-    }
-
-    const onPrevious = () => {
-      onChangePage(currentPage - 1)
-    }
-
-    const onNext = () => {
-      onChangePage(currentPage + 1)
-    }
-
-    const navLabelId = useId()
+    const range = getRange(page, totalPages, maxVisiblePages)
 
     // Don't show pagination if you only have one page
     if (totalPages <= 1) {
       return null
     }
 
     return (
```

Comment:
The render function still looks pretty overwhelming. I think extracting local `PreviousLink`, `PageLink`, `Spacer`, and `NextLink` components would clarify.

Reply:
As discussed, I've extracted `LinkItem` and `Spacer`. `PreviousLink` and `NextLink` would not add much.